parachain/pallets/bridge/src/lib.rs

```diff
@@ -85,54 +104,26 @@ decl_module! {
 		fn deposit_event() = default;
 
 		#[weight = 0]
-		pub fn submit(origin, channel_id: ChannelId, app_id: AppId, message: Message) -> DispatchResult {
+		pub fn submit(origin, message: Message) -> DispatchResult {
 			let relayer = ensure_signed(origin)?;
 
-			let mut channel = make_inbound_channel::<T>(channel_id);
-			channel.submit(&relayer, app_id, &message)
-		}
-
-		/// Submit multiple messages for dispatch to multiple target applications.
-		#[weight = 0]
-		pub fn submit_bulk(origin, messages: Vec<(AppId, Message)>) -> DispatchResult {
-			let who = ensure_signed(origin)?;
-
-			debug::RuntimeLogger::init();
-
-			debug::trace!(
-				target: "submit_messages",
-				"Received {} messages",
-				messages.len(),
-			);
-
-			T::Verifier::verify_bulk(who, messages.as_slice())?;
-
-			debug::trace!(
-				target: "submit_messages",
-				"Message verification succeeded",
-			);
-
-			let errors: Vec<DispatchError> = messages.iter()
-				.map(|(app_id, msg)| Self::dispatch(app_id.into(), msg))
-				.filter_map(|r| r.err())
-				.collect();
-
-			debug::trace!(
-				target: "submit_messages",
-				"Messages were dispatched",
-			);
+			let log = T::Verifier::verify(&message)?;
+			let envelope = Envelope::try_from(log).map_err(|_| Error::<T>::InvalidEnvelope)?;
+			let channel_id = SourceChannels::get(envelope.channel)
```

Comment:
channel_id is the contract address of the corresponding outbound channel right?

Reply:
Is a bit more complicated. the `SourceChannels` storage item is a mapping between a ChannelId and a corresponding address for a ethereum outbound channel.

```
pub enum ChannelId {
	Basic,
	Incentivized
}
```